procedures/automation-android-setup.md

```diff
@@ -0,0 +1,200 @@
+# How to set up automation for nightly Android product
+
+These instructions define which resources are needed for setting up nightly Android product release automation, as well as the steps to prepare the automation itself. Additionally, this guide include steps for running a "staging" version of the automation so that it can be tested in the developer's own repository.
+
+Throughout this document, wherever the term `$product` is used, substitute your product's name in (replacing spaces with hyphens), e.g.: `reference-browser` or `fenix`.
+
+Note: we don't need to explicitly define scopes in Taskcluster. We just define them in the task, hook's role, and each scriptworker script. Taskcluster will implicitly detect scopes, but it doesn't use them for anything other than generic equality/wildcard verification.
+
+1. Request dep and release signing keys (example bug for [`reference-browser`](https://bugzilla.mozilla.org/show_bug.cgi?id=1508761)). Confirm with the app's team what the "signature algorith" and "digest algorithm" should be, and include that information in the ticket.
+2. Clone the product's repository
+    * Note that, while doing "staging" runs, you'll have to use your `master` branch in your fork. Hook cloning/CoT/etc haven't been tested with a non-default branch
+3. Add `.taskcluster.yml` in the root of the repository if it doesn't exist, and add a task for cron jobs. ([`fenix` example, go to `tasks_for == "cron"`](https://github.com/mozilla-mobile/fenix/blob/master/.taskcluster.yml))
+    1. Update the `event.repository.clone_url` checks to use the correct repo URL
```

Comment:
We may want to port the logic that we have in android-components to make this step seamless on other repos. Pointers: 

* https://github.com/mozilla-mobile/android-components/blob/bd045b5d820edeb05dfef3e254e2c5a0b5261d4a/.taskcluster.yml#L32-L34
* https://github.com/mozilla-mobile/android-components/blob/bd045b5d820edeb05dfef3e254e2c5a0b5261d4a/.taskcluster.yml#L102
* https://github.com/mozilla-mobile/android-components/blob/bd045b5d820edeb05dfef3e254e2c5a0b5261d4a/.taskcluster.yml#L111
* https://github.com/mozilla-mobile/android-components/blob/bd045b5d820edeb05dfef3e254e2c5a0b5261d4a/.taskcluster.yml#L232

Reply:
I don't quite follow: 

* Are you referring to making `$repository` work for both hook and pull request triggers? In this case, since the documentation is specifically about setting up nightly, I'd like to opt towards leaving it simple and let a follow-up guide explain how to set up PR checks :)
* Alternatively, ado you mean how we need to have checks that the correct repo URL is used, and how we're directly referring to the repository (`https://github.com/mozilla-mobile/$project`)? I'm not sure that we can work around that, and `android-components` [has the same thing](https://github.com/mozilla-mobile/android-components/blob/bd045b5d820edeb05dfef3e254e2c5a0b5261d4a/.taskcluster.yml#L44) to check if the repo is trusted.

TL;DR I'm a nerd and I don't understand the recommendation :)

Reply:
I'm sorry. Even I was confused when I read my comment. I mean: if we extract the repository from the Github event and we pass it down to `decision.py`, there is no need to ``event.repository.html_url` at all. It just works without having to temporary modify some params.

Reply:
Sounds good: If you're on-board, I want the docs to point to "Fenix" as the reference for automation setup, because:
1. I'm biased, I set it up :stuck_out_tongue: 
2. `android-components` has more mature automation, but it has the complexity of a variable number of sub-projects (and it doesn't do pushapk). I'd rather that our reference be more like a standalone product
3. `reference-browser` is less production-y than Fenix

When we update Fenix's `decision.py` and how it works, let's update the docs, too!